cvc5/Solver.lean

```diff
@@ -314,8 +385,81 @@ private opaque mkIntegerFromString : TermManager → String → Term
 def mkInteger (tm : TermManager) : Int → Term :=
   (mkIntegerFromString tm) ∘ toString
 
+/-- Create operator of Kind:
+
+- `Kind.BITVECTOR_EXTRACT`
+- `Kind.BITVECTOR_REPEAT`
+- `Kind.BITVECTOR_ROTATE_LEFT`
+- `Kind.BITVECTOR_ROTATE_RIGHT`
+- `Kind.BITVECTOR_SIGN_EXTEND`
+- `Kind.BITVECTOR_ZERO_EXTEND`
+- `Kind.DIVISIBLE`
+- `Kind.FLOATINGPOINT_TO_FP_FROM_FP`
+- `Kind.FLOATINGPOINT_TO_FP_FROM_IEEE_BV`
+- `Kind.FLOATINGPOINT_TO_FP_FROM_REAL`
+- `Kind.FLOATINGPOINT_TO_FP_FROM_SBV`
+- `Kind.FLOATINGPOINT_TO_FP_FROM_UBV`
+- `Kind.FLOATINGPOINT_TO_SBV`
+- `Kind.FLOATINGPOINT_TO_UBV`
+- `Kind.INT_TO_BITVECTOR`
+- `Kind.TUPLE_PROJECT`
+
+See `cvc5.Kind` for a description of the parameters.
+
+- `kind` The kind of the operator.
+- `args` The arguments (indices) of the operator.
+
+If `args` is empty, the `Op` simply wraps the `cvc5.Kind`. The `Kind` can be used in `Solver.mkTerm`
+directly without creating an `Op` first.
+-/
+@[extern "termManager_mkOpOfIndices"]
+opaque mkOpOfIndices : TermManager → (kind : Kind) → (args : Array Nat) → Op
+
+/-- Create operator of kind:
+
+- `Kind.DIVISIBLE` (to support arbitrary precision integers)
+
+See `cvc5.Kind` for a description of the parameters.
+
+- `kind` The kind of the operator.
+- `arg` The string argument to this operator.
+
+-/
+@[extern "termManager_mkOpOfString"]
+private opaque mkOpOfString : TermManager → (kind : Kind) → (arg : String) → Op
+
+/-- Create divisibility-by operator, supports arbitrary precision. -/
+def mkOpDivisible (tm : TermManager) (n : Nat) (_valid : 0 < n := by simp) : Op :=
+  tm.mkOpOfString Kind.DIVISIBLE (toString n)

```

Comment:
I don't think we're handling this function properly (it should also support rationals)... let's delete for now...